src/liquid/components/ld-select/ld-select.tsx

```diff
@@ -750,20 +791,25 @@ export class LdSelect implements InnerFocusable {
         }
         break
       case 'Tab': // Also covers Shift+Tab
-        // If expanded: Prevent default.
-        if (this.expanded) {
+        // If expanded and popper element has focus within: Prevent default.
+        if (
+          this.expanded &&
+          document.activeElement.closest('[role="listbox"]') === this.listboxRef
+        ) {
           ev.preventDefault()
           ev.stopImmediatePropagation()
         }
         break
-      case 'Shift':
-      case 'Meta':
-        break
       default:
         if (this.expanded) {
-          ev.preventDefault()
           ev.stopImmediatePropagation()
-          this.typeAhead(ev.key)
+          if (
+            !this.filter ||
+            this.el.shadowRoot.activeElement !== this.getFilterInput()
+          ) {
+            ev.preventDefault()
+            this.typeAhead(ev.key)
+          }
```

Comment:
Somewhere inside this function is code that results in inconsistent behavior. When focus is inside popper, hitting [Shift] results in the focus jumping somewhere to the top. Not always the first element. This may not be new behavior, but this method becomes so complex that I think it should be simplified (e.g. the code for different cases being moved to separate functions), and maybe refactored to get a more consistent behavior.

Reply:
I removed code, because I didn't know anymore what it was useful for, but now I know it again:
```
case 'Shift':
case 'Meta':
        break
```
I will add a comment.

Reply:
Actually I will move this into the typeAhead function, so that it is more obvious what is being prevented there.

Reply:
Good catch by the way!